src/__tests__/helpers.ts

```diff
@@ -3,12 +3,21 @@ import {
     Serializer,
     serializers as predefinedSerializers,
 } from '../';
+import { resetUuidCounter } from './setup';
 
-type SerializerType = 'json' | 'yaml' | 'dot' | 'structure';
+type SerializerType =
+    | 'json'
+    | 'json_depends_on'
+    | 'yaml'
+    | 'yaml_depends_on'
+    | 'dot'
+    | 'structure';
```

Comment:
I think rather than adding new types for this, and have a combinatorial explosion of feature * output types, this should be handled separately.

See https://docs.google.com/document/d/1YHEVvW9IPhzszgp3e0vKITQubKmAZyO5F_DA28SD9jU/edit# for example.

Reply:
I did it so I could get separate tests. Can introduce the naming for tests only and remove the specialised serializers. I removed to doc link.